src/os/bluestore/BlueStore.cc

```diff
@@ -5467,18 +5512,47 @@ int BlueStore::_init_alloc()
 #endif
   
   uint64_t num = 0, bytes = 0;
+  utime_t start_time = ceph_clock_now();
+  // This is the original path - loading allocation map from RocksDB and feeding into the allocator
+  if (!fm->is_null_manager()) {
+    dout(5) << __func__ << "::NCB::loading allocation from FM -> shared_alloc" << dendl;
+    // initialize from freelist
+    fm->enumerate_reset();
+    uint64_t offset, length;
+    while (fm->enumerate_next(db, &offset, &length)) {
+      shared_alloc.a->init_add_free(offset, length);
+      ++num;
+      bytes += length;
+    }
+    fm->enumerate_reset();
+
+    utime_t duration = ceph_clock_now() - start_time; 
+    dout(5) << __func__ << "::num_entries=" << num << " free_size=" << bytes << " alloc_size=" <<
+      shared_alloc.a->get_capacity() - bytes << " time=" << duration << " seconds" << dendl;
+  } else {
+    // This is the new path reading the allocation map from a flat bluefs file and feeding them into the allocator
+    
+    if (!cct->_conf->bluestore_allocation_from_file) {
+      derr << __func__ << "::NCB::cct->_conf->bluestore_allocation_from_file is set to FALSE with an active NULL-FM" << dendl;
+      derr << __func__ << "::NCB::Please change the value of bluestore_allocation_from_file to TRUE in your ceph.conf file" << dendl;
+      exit(0);
```

Comment:
Something is clearly wrong here and we should not progress

Reply:
exit(0) is not correct - it will hide the issue since the exit code is success. returning an error code here would be better - it ends up cleaning things up giving an error to user that OSD init failed